### DISCUSSION TITLE
Transforming the output to enforce a sharp (discontinuity) initial condition

### DISCUSSION THREAD

**Discussion by valderrama-juan** [lululxvi/deepxde#1332] [Q&A]
Hi @lululxvi !

I am currently working on building a PINN that learns the diffusion front layer for an applied physics problem. While there are some obscure details about the diffusion coefficients dictating the mixing layer behavior, I have managed to implement them well into the solution of the problem. My issue lies with enforcing the initial and boundary conditions.

The initial condition of the problem resembles a "step function" of the form y=(tanh(x/delta_x)+1)/4, where delta_x is a value that I want to make as small as possible. (i.e., a function with range [0,0.5] that "jumps" at x=0). The boundary conditions are y=0 at the left boundary (i.e., x=-1) and y=0.5 at the right boundary (i.e., x=1). 

To enforce the initial and boundary conditions as hard constraints, I have created the following output transform:

`
delta_x = 1.e-1

def output_transform(inputs, outputs):

    x, t = inputs[:, 0:1], inputs[:, 1:2]

    y_guess = (tf.tanh(x/delta_x)+1)/4

    bc = t*(xMax-x)*(x-xMin)

    y = y_guess + bc * outputs

    return y`

In this manner, the initial condition y=(tanh(x/delta_x)+1)/4 is returned when t=0, and the boundary conditions are returned: y(x=-1,t)=0, y(x=1,t)=0.5.

I can confirm that the loss obtained for this transform is desirable (i.e., loss=4.89e-07), and the network trains successfully. However, I desire to make the value of delta_x smaller to better reflect the "discontinuity" of the initial condition. When I do this (e.g., set delta_x=1.e-2) the network fails to train and the solution is no longer what I desire. 

My question is then, how can I implement the initial condition differently so that the network still trains when making delta_x smaller? Is there a mistake with the way in which I am expressing the output transform? I can provide more details if needed.

**Comment by lululxvi** [comment]
Neural networks are always continuous. Try other activations like selu or swish.
